### PR TITLE
Update the xcode cloud snippet

### PR DESCRIPTION
**Pull request by farcaller**
_Description of what this PR is changing or adding, and why:_

Only do a shallow clone for flutter as it's faster.

_Issues fixed by this PR (if any):_

N/A

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/deployment/cd.md [hunk 1]
```diff
@@ -327,4 +327,4 @@ information.
 [Xcode Cloud workflow]: https://developer.apple.com/documentation/xcode/xcode-cloud-workflow-reference
 [custom build scripts]: https://developer.apple.com/documentation/xcode/writing-custom-build-scripts
 [predefined environment variables]: https://developer.apple.com/documentation/xcode/environment-variable-reference
-[Setting the next build number for Xcode Cloud builds]: https://developer.apple.com/documentation/xcode/setting-the-next-build-number-for-xcode-cloud-builds#Set-the-next-build-number-to-a-custom-value
\ No newline at end of file
+[Setting the next build number for Xcode Cloud builds]: https://developer.apple.com/documentation/xcode/setting-the-next-build-number-for-xcode-cloud-builds#Set-the-next-build-number-to-a-custom-value
```
Comment (domesticmouse) [inline src/deployment/cd.md:330]: What is the difference between these two lines? I'm guessing addition or removal of a trailing newline at EOF?

Comment (farcaller) [reply src/deployment/cd.md:330]: Yeah, it's a trailing EOF I’m afraid (did it via the GH online editor). I can clean it up if needed.

File: src/deployment/cd.md [hunk 2]
```diff
@@ -231,7 +231,7 @@ Create a file at `ios/ci_scripts/ci_post_clone.sh` and add the content below.
 cd $CI_WORKSPACE # change working directory to the root of your cloned repo.
 
 # Install Flutter using git.
-git clone https://github.com/flutter/flutter.git -b stable $HOME/flutter
+git clone https://github.com/flutter/flutter.git --depth 1 -b stable $HOME/flutter
```
Comment (domesticmouse) [inline src/deployment/cd.md:234]: This is snippet'd in from `deployment/xcode_cloud/ci_post_clone.sh`. Please modify the source and use the tool referenced in the README to update the snippets. The CI will fail this as it stands.

**Review by domesticmouse — CHANGES_REQUESTED** [review]

File: src/deployment/cd.md (hunk 1 above)
Comment (domesticmouse) [reply src/deployment/cd.md:330]: Ahh, cool. I'm happy for this file to have a trailing new line. =)

File: src/deployment/cd.md (hunk 2 above)
Comment (farcaller) [reply src/deployment/cd.md:234]: I think it should work proper now (the fun times trying to run the tooling under windows, ha).

**Review by domesticmouse — APPROVED** [review]

**Pull request merged**